pylint/checkers/classes.py

```diff
@@ -2299,6 +2303,17 @@ def _is_iterator(node):
                     pass
         return False
 
+    @staticmethod
+    def _function_body_is_ellipsis(node: nodes.FunctionDef) -> bool:
```

Comment:
What do you think about moving this method to `checkers/utils.py`?

Reply:
Shouldn't it be in astroid's FunctionDef instead ?

Reply:
> Shouldn't it be in astroid's FunctionDef instead ?

Though about this some more. I'm not sure we should move these kinds of helper functions to `astroid`. `astroid` only really deals with the `ast` and inference, everything else should be done by `pylint`.

By the same logic I believe it might have been wrong to move `is_sys_guard` and `is_typing_guard` to `astroid` in the first place. They don't really belong there.

Reply:
Couldn't it be argued that these methods are part of the inference? In that they give additional information about the specific type of `FunctionDef` node?
Anyway, I'll leave this up to you and Pierre. I'm fine with either. 
If wanted I can also make a PR to re-move `is_sys_guard` and `is_typing_guard` to `pylint`.

Reply:
> Couldn't it be argued that these methods are part of the inference? In that they give additional information about the specific type of `FunctionDef` node?

True. Personally I would consider this a case of meta inference. We check multiple conditions to decide wether or not to emit a warning. That's that sort of thing `pylint` does all the time.

One thing to consider with that, it's much easier to iterate (read: fix) such a check in `pylint` alone than to modify something in `astroid` + `pylint`.

What I do consider useful in `astroid` are helpers like this one https://github.com/PyCQA/astroid/pull/1169 which helps dealing with the `NodeNG` tree itself.


> Anyway, I'll leave this up to you and Pierre. I'm fine with either.
> If wanted I can also make a PR to re-move `is_sys_guard` and `is_typing_guard` to `pylint`.

I would be in favor of it, but let's see what @Pierre-Sassoulas thinks first. Unfortunately, we can't outright remove them now, so we would need to deprecate them first (and remove with astroid 3.0).

Reply:
I think your second argument might tip the scale (it does for me anyway). Both places seem reasonable but the benefit of being able to tweak these checks by only updating `pylint` is indeed quite good.
If Pierre agrees I would:
1. Fix this PR from code review while keeping method in `pylint`
2. Create PR which re-adds `is_sys_guard` and `is_typing_guard` to `pylint`
3. Create deprecating PR for `is_sys_guard` and `is_typing_guard` to be merged after `2` is merged into a release of `pylint` (so users can actually use the new methods)

Reply:
Sounds good to me! As it's only a deprecation `3` wouldn't necessarily need to wait for `2`.
In principle I would say all of the helpers mentioned should probably be part of `checkers/utils.py` (if no circular dependency gets in the way).

Reply:
It make sense. It's easy to see that when something rely on astroid any pylint modification slow down by a factor of 10. Could we find a middle ground and not remove what is already in astroid but not add anything either ? It's less work :smile: (for us but also for dev using astroid as a library)

Reply:
I think the middle ground just make the codebase inconsistent. We already have some deprecation warnings in both `astroid` and `pylint` for a next major release so we need to take a look at these anyway (so the extra work is relatively minimal). 
Furthermore, by doing step 3 after step 2 we can direct users to the "new" `pylint` methods if they want to keep using them. That would require minimal work for them as well.

Reply:
Well you're right about consistency but there are developpers using astroid without using pylint, removing this from astroid could be problematic if they started using the typing guard in their code. Also, I think the main problem is for project that expects any pylint to work with any astroid and are setting astroid manually (prospector being one of them until very recently). Maybe it will be the occasion for them to let pylint handle its own dependencies.

Reply:
So, we would leave the situation as is?

Reply:
I'm not advocating for one solution or the other, consistency is important, we just need to consider the other factors. What do you think @cdce8p ?

Reply:
I would probably prefer to move these back to pylint, for consistency. If we don't do it, someone else might come along and think it's accepted to add other methods there as well.

Furthermore, as we've only added it recently, there might only be a few projects that use it. Most of them can import it from `pylint` without issues.

Reply:
Do you mean first release `2.8.1` and afterwards deprecate, or the other way around?

Reply:
I guess, let's do the deprecation now. I.e. wait with releasing astroid `2.8.1` until it's done.
https://github.com/PyCQA/astroid/issues/1199

Reply:
I opened https://github.com/PyCQA/astroid/pull/1202

Reply:
Opened #5122 to add the guard helper functions back to pylint.